src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2905,7 +2905,7 @@ private static void GetAllCandidates(Dictionary<TypeSymbolWithAnnotations, TypeS
                     {
                         candidates.Remove(candidate);
                     }
-                    else if (bound.Equals(candidate, TypeCompareKind.IgnoreDynamicAndTupleNames))
+                    else if (bound.Equals(candidate, TypeCompareKind.IgnoreDynamicAndTupleNames | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
```

Comment:
This seems to be used relatively often. Might be worth adding an explicit option in the enum.